### PR TITLE
Feature make goal

### PR DESCRIPTION
**Pull request by mbabey**
Merge the culmination of changes in branches 'feature-edit-bio,' 'feature-bio-pop-info,' and 'feature-make-goal.' Includes functionality for changing user profile info and making new goals with database integration.

### CODE REVIEW

File: icons/edit-2.svg [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (nok-ko) [inline icons/edit-2.svg:1]: Why the blank line?

File: scripts/profile.js [hunk 2]
```diff
@@ -53,11 +53,156 @@ function listGoals() {
...
+		// Check if user is signed in:
+		if (user) {
+			//Assign UID to currentUser.
+			currentUser = db.collection("users").doc(user.uid);
+
+			//go to the correct user document by referencing to the user uid
+			//get the document for current user.
+			currentUser.onSnapshot(userDoc => {
+				//get the data fields of the user
+				var userName = userDoc.data().name;
```
Comment (nok-ko) [inline scripts/profile.js:71]: Try to use `let` instead of `var`, helps avoid weird one-off scoping issues.

File: icons/edit-2.svg (hunk 1 above)
Comment (mbabey) [reply icons/edit-2.svg:1]: Was editing it to try to make the SVG appear in profile.html and must have missed this.

File: scripts/profile.js (hunk 2 above)
Comment (mbabey) [reply scripts/profile.js:71]: rog

File: scripts/profile.js [hunk 3]
```diff
@@ -53,11 +53,156 @@ function listGoals() {
...
+		console.log("Name updated successfully.")
+	})
+
+	//Disable the form fields. Makes the input box invisible and
+	//	turn the name text on.
+	document.getElementById('nameField').disabled = true;
+	document.getElementById('nameInput').hidden = true;
+	document.getElementById('nameText').hidden = false;
+	//Disappear the save version of the button and appear the edit version of the button.
+	//Not currently working.
```
Comment (nok-ko) [inline scripts/profile.js:133]: Interesting – maybe add a card & fix later.

**Review by nok-ko — APPROVED** [review]
Looks good to me!

File: icons/edit-2.svg (hunk 1 above)
Comment (nok-ko) [reply icons/edit-2.svg:1]: I see, all good. We'll catch that kind of thing in a cleanup pass later.

**Pull request merged**